### PR TITLE
Feature/sim 1858 replace junit timeout annotation property with asserttimeout

### PR DESCRIPTION
**Pull request by arditymeri**

### CODE REVIEW

File: docs/rules/replace-j-unit-timeout-annotation-property.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+@Test
+public void multipleStatements() throws PersistenceException {
+	assertTimeout(ofMillis(100), () -> {
+		User user = new User("10", "John", "Snow");
+		userRepository.save(user);
+	});
+}
+```
+
+### Multiple Annotation Properties Properties
```
Comment (gregor-balint) [inline docs/rules/replace-j-unit-timeout-annotation-property.md:82]: Please delete second 'Properties'

File: docs/rules/replace-j-unit-timeout-annotation-property.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+
+::: warning Requirements
+This rule requires the following library to be present:
+* org.junit.jupiter:junit-jupiter-api:5.0.0
+:::
+
+## Benefits
+
+Improves the tests readability. Helps with migrating to JUnit 5.
```
Comment (gregor-balint) [inline docs/rules/replace-j-unit-timeout-annotation-property.md:36]: I would write:
Improves readability of tests. Helps migrating to JUnit 5.

File: docs/rules/replace-j-unit-timeout-annotation-property.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+[[toc]]
+
+## Properties
+
+<RuleProperties />
+
+## Description
+
+JUnit [Jupiter API](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html) provides timeout assertions, i.e., assertions that execution of some code completes before a timeout exceeds.
```
Comment (gregor-balint) [inline docs/rules/replace-j-unit-timeout-annotation-property.md:23]: I would change the link [Jupiter API](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertTimeout-java.time.Duration-org.junit.jupiter.api.function.Executable-) which shows the first assertTimeout - method

Comment (gregor-balint) [inline docs/rules/replace-j-unit-timeout-annotation-property.md:23]: I would change the content of the link to: 
`https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertTimeout-java.time.Duration-org.junit.jupiter.api.function.Executable-`

Comment (gregor-balint) [inline docs/rules/replace-j-unit-timeout-annotation-property.md:23]: IMO, in line 23 the link to the overwiew with
`[Jupiter API](https://junit.org/junit5/docs/5.0.1/api/overview-summary.html)`
would be sufficient.

File: docs/rules/replace-j-unit-timeout-annotation-property.md (hunk 1 above)
Comment (arditymeri) [reply docs/rules/replace-j-unit-timeout-annotation-property.md:82]: Nice catch

**Pull request merged**